packages/btp-utils/src/app-studio.ts

```diff
@@ -127,3 +149,128 @@ export async function exposePort(port: number, logger?: Logger): Promise<string>
+ * @param logger Logger
+ * @returns Preconfigured OAuth destination
+ */
+async function generateOAuth2UserTokenExchangeDestination(
+    destination: Destination,
+    serviceInstanceName: string,
+    logger?: Logger
+): Promise<destinations.DestinationInfo> {
+    if (!serviceInstanceName) {
+        throw new Error(`No service instance name defined.`);
+    }
+
+    const destinationName: string = await generateABAPCloudDestinationName(destination.Name);
+    const instances: CloudFoundryServiceInfo[] = await apiGetServicesInstancesFilteredByType(['destination']);
+    const destinationInstance = instances.find(
+        (instance: CloudFoundryServiceInfo) => instance.label === DESTINATION_INSTANCE_NAME
+    );
+
+    if (!destinationInstance) {
+        // Create a new abap-cloud destination instance on the target CF subaccount
+        await apiCreateServiceInstance('destination', 'lite', DESTINATION_INSTANCE_NAME, null);
+        logger?.info(`New ABAP destination instance ${DESTINATION_INSTANCE_NAME} created.`);
+    }
+
+    const instanceDetails = await apiGetInstanceCredentials(serviceInstanceName);
+    if (!instanceDetails?.credentials) {
+        throw new Error(`Could not retrieve SAP BTP credentials for ${serviceInstanceName}.`);
+    }
+    return transformToBASSDKDestination(
+        {
+            ...destination,
+            Description: `Destination generated by App Studio for '${destination.Name}', Do not remove.`,
+            Name: destinationName
+        },
+        instanceDetails.credentials as ServiceInfo['uaa']
+    );
+}
+
+/**
+ *  Create a new SAP BTP subaccount destination of type 'OAuth2UserTokenExchange' using cf-tools to populate the UAA properties.
+ *  If the destination already exists, only new or missing properties will be appended, existing fields are not updated with newer values.
+ *
+ *  For example: If an existing SAP BTP destination already contains `WebIDEEnabled` and the value is set as `false`, the value will remain `false` even after the update.
+ *
+ * @param destination destination info
+ * @param serviceInstanceName name of the service instance, for example, the ABAP Environment service name reflecting name of the service created using a supported service technica name i.e. abap | abap-canary
+ * @param logger Logger
+ * @returns name representing the newly created SAP BTP destination
+ */
+export async function createOAuth2UserTokenExchangeDest(
+    destination: Destination,
+    serviceInstanceName: string,
+    logger?: Logger
+): Promise<string> {
+    if (!isAppStudio()) {
+        throw new Error(`Creating a SAP BTP destinations is only supported on SAP Business Application Studio.`);
+    }
+    const basSDKDestination: destinations.DestinationInfo = await generateOAuth2UserTokenExchangeDestination(
+        destination,
+        serviceInstanceName,
+        logger
+    );
+    await destinations.createDestination(basSDKDestination);
```

Comment:
Can you instead return the new destination? This is a strange tightly coupled interaction otherwise.